tests/unit/test_charm.py

```diff
@@ -1785,11 +1785,7 @@ def test_get_secret_secrets(harness, scope, field):
 
 
 @patch_network_get(private_address="1.1.1.1")
-def test_set_secret_in_databag(harness, _has_secrets):
-    # As this test set secrets using set_secret and checks if they end up
-    # inside the relation databag, this is meant to run on juju2 only
-    if _has_secrets:
-        return
+def test_set_secret_in_databag(harness, only_without_juju_secrets):
```

Comment:
We should keep the comment it was very good :-)

Reply:
Those comments that you had 1-2 commits ago were great. I think it's a very good idea to add a small explanation (docstring?) to these tests. I left them pretty confusing :-(

You don't have to add them but if you choose to do they will be helpful :-)